### PR TITLE
Apply rotary kernel

### PR DESCRIPTION
**Pull request by grimoire**
- add apply rotary kernel
- add rms norm kernel
- unittest

### CODE REVIEW

File: tests/pytorch/kernel/test_rms_norm.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+class TestRMSNorm:
+
+    @pytest.fixture(scope='class')
+    def input(self):
+        yield torch.rand(4, 8, dtype=torch.float16, device='cuda')
+
+    @pytest.fixture(scope='class')
+    def weight(self):
+        yield torch.rand(8, dtype=torch.float16, device='cuda')
```
Comment (wangruohui) [inline tests/pytorch/kernel/test_rms_norm.py:13]: I see dtypes are handled in the kernel, how about testing with some bfloat16/float32 dtype?

File: lmdeploy/pytorch_poc/models/functional.py [hunk 2]
```diff
@@ -7,51 +7,10 @@
...
-        seq_length = position_ids[..., -1] + 1
-        position_ids_1d = [ids[:l] for ids, l in zip(position_ids, seq_length)]
-        position_ids_1d = torch.cat(position_ids_1d)
-    cos = cos[position_ids_1d].unsqueeze(1)
-    sin = sin[position_ids_1d].unsqueeze(1)
-    q_embed = (q * cos) + (rotate_half(q) * sin)
-    k_embed = (k * cos) + (rotate_half(k) * sin)
-
-    return q_embed, k_embed
+apply_rotary_pos_emb = apply_rotary_pos_emb
```
Comment (wangruohui) [inline lmdeploy/pytorch_poc/models/functional.py:13]: left are same as right?

File: tests/pytorch/kernel/test_rms_norm.py (hunk 1 above)
Comment (grimoire) [reply tests/pytorch/kernel/test_rms_norm.py]: Done

File: lmdeploy/pytorch_poc/models/functional.py (hunk 2 above)
Comment (grimoire) [reply lmdeploy/pytorch_poc/models/functional.py]: Replaced with `__all__=['apply_rotary_pos_emb']`

**Review by wangruohui — APPROVED** [review]

**Review by AllentDan — APPROVED** [review]

**Pull request merged**